### ISSUE TITLE
Error during releasing of frozen database with Lucene indexes

### ISSUE THREAD

**Issue by antonzherdev** [orientechnologies/orientdb#6733]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .9[X] .rc1[ ] .beta2[ ] .beta1[ ]

## Operating System
- [X] Linux
- [ ] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

## Expected behavior and actual behaviour
Error:
```
2016-09-21 15:00:19,337 [OrientDB (/127.0.0.1:2424) <- BinaryClient (/127.0.0.1:53326)] INFO  c.o.o.s.n.p.b.ONetworkProtocolBinary [27.0.0.1:53326)] - $ANSI{
green {db=wdb}} Realising database 'plocal:/var/erpdb/wdb'
2016-09-21 15:00:19,338 [OrientDB (/127.0.0.1:2424) <- BinaryClient (/127.0.0.1:53326)] ERROR c.o.l.e.OLuceneFullTextIndexEngine [27.0.0.1:53326)] - $ANSI{gr
een {db=wdb}} Error on releasing Lucene index
org.apache.lucene.store.LockObtainFailedException: Lock held by this virtual machine: /var/erpdb/wdb/luceneIndexes/ft_13_8/write.lock
        at org.apache.lucene.store.NativeFSLockFactory.obtainFSLock(NativeFSLockFactory.java:127)
        at org.apache.lucene.store.FSLockFactory.obtainLock(FSLockFactory.java:41)
        at org.apache.lucene.store.BaseDirectory.obtainLock(BaseDirectory.java:45)
        at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:775)
        at com.orientechnologies.lucene.engine.OLuceneIndexWriterFactory.createIndexWriter(OLuceneIndexWriterFactory.java:22)
        at com.orientechnologies.lucene.engine.OLuceneFullTextIndexEngine.createIndexWriter(OLuceneFullTextIndexEngine.java:73)
        at com.orientechnologies.lucene.engine.OLuceneIndexEngineAbstract.reOpen(OLuceneIndexEngineAbstract.java:210)
        at com.orientechnologies.lucene.engine.OLuceneIndexEngineAbstract.release(OLuceneIndexEngineAbstract.java:506)
        at com.orientechnologies.lucene.engine.OLuceneIndexEngineDelegate.release(OLuceneIndexEngineDelegate.java:278)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.release(OAbstractPaginatedStorage.java:2629)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.release(ODatabaseDocumentTx.java:2375)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.releaseDatabase(ONetworkProtocolBinary.java:2125)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:601)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sessionRequest(ONetworkProtocolBinary.java:318)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:194)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

Database stays frozen after this.

## Steps to reproduce the problem
It is not stable but very critical:
1. Freeze a database with Lucene indexes
2. Release it

**Comment by maggiolo00** [comment]
hi @antonzherdev 

do you manually invoke freeze/release from console?

**Comment by antonzherdev** [comment]
Hi @maggiolo00 
No, it's a script which starts periodically. I'm freezing the database, making AWS snapshot and releasing it. Sometimes it's working fine, sometimes I'm getting this problem. The script looks like:

```
CONSOLE="sh /opt/orientdb/bin/console.sh"

$CONSOLE "connect REMOTE:/localhost/wdb admin $DB_PASSWORD; freeze database; disconnect;"

aws ec2 create-snapshot --volume-id ...

$CONSOLE "connect REMOTE:/localhost/wdb admin $DB_PASSWORD; release database; disconnect;"
```

**Comment by maggiolo00** [comment]
@antonzherdev 

ok i'm checking it

**Comment by maggiolo00** [comment]
hi @antonzherdev 

i've reproduced your stack error only if i call two times release database.
I will continue digging

**Comment by maggiolo00** [comment]
@antonzherdev 

do you run in standalone or distributed mode?

**Comment by antonzherdev** [comment]
@maggiolo00 

In standalone. Snapshot can happen very fast. So there could be some freeze operations which are not finished yet. 

I suppose even if release is called twice it should not give this stack. In this case there should be an error that the database is not frozen, or do you mean two releases in parallel?

**Comment by maggiolo00** [comment]
@antonzherdev 
i've pushed a fix. It will be available in 2.2.11.

**Comment by antonzherdev** [comment]
@maggiolo00 Thank you so much!
